### ISSUE TITLE
Coreupdater issues

### ISSUE THREAD

**Issue by musicpanda** [thirtybees/coreupdater#20]
I tried Coreupdater with two testshops and both gave problems. The PHP version I used was 7.3 and in both cases the installation forced me to upgrade the Coreupdater to the latest version (1.45).

One installation (version 1.0.8) produced this error: <b>Deprecated</b>:  The each() function is deprecated. This message will be suppressed on further calls in <b>www.shop.com\vendor\smarty\smarty\libs\sysplugins\smarty_internal_compilebase.php</b> on line <b>75</b><br />
![ajax](https://user-images.githubusercontent.com/13930107/146986518-381288d8-73ee-49e7-8601-f9b26c114dfd.jpg)

The second installation (version 1.2) gave more serious problems. In this case the clicking on Configure in the Modules page produced this:
![core12](https://user-images.githubusercontent.com/13930107/146987086-9c4a451f-7cf7-48bf-9b0e-99b09d3575d0.jpg)

There is here no way to see what versions are available for upgrade. It is just a static page where you can save some settings.

Going in the menu to Settings->Coreupdater I got an empty screen:
![empty](https://user-images.githubusercontent.com/13930107/146987562-3423dea6-e15f-4aa3-b8a8-f96643d20c99.jpg)

When I look in the console I see a javascript error: Uncaught ReferenceError: initializeCoreUpdater is not defined on line 891 of admin/index.php?controller=AdminCoreUpdater
![reference](https://user-images.githubusercontent.com/13930107/146988154-8400b9b4-b7c8-463c-ab66-f37f7fa49902.jpg)

**Comment by getdatakick** [comment]
The first two issues seems duplicate of issues #18 and #17.

Error handling -- core updater module does not come with it own error handling, it depends on error handling provided by underlying thirty bees core. That works nicely if core is 1.1.0. Unfortunately, on older tb (1.0.8), the error handling/reporting in the core is not very good. PHP warning/notices can be emitted into the output. For html code it's not critical, however for json response this has very serious consequences, as javascript is unable to parse such output.

As for cors issue -- again, core updater use core code to generate link for ajax endpoints. Some older core versions can return invalid url, from CORS perspective

Basically, what we need to do is get rid of dependencies on core, as much as possible. Ideally, core updater should be able to run as a standalone web app

**Comment by getdatakick** [comment]
The last issue - empty list of changed files - is fixed by commit 934898497ac13b8d5e26c62e5a1fbdbf47c32030
